ethclient/ethclient.go

```diff
@@ -150,7 +150,10 @@ func (ec *Client) getBlock(ctx context.Context, method string, args ...interface
 	if head.UncleHash == types.EmptyUncleHash && len(body.UncleHashes) > 0 {
 		return nil, errors.New("server returned non-empty uncle list but block header indicates no uncles")
 	}
-	if head.UncleHash != types.EmptyUncleHash && len(body.UncleHashes) == 0 {
+	// In celo before the ginerbread hardfork, blocks had no uncles hash and no
+	// uncles, so in those cases it is legitimate to have an empty uncles hash.
+	var emptyHash common.Hash
```

Comment:
Should we allow this only in a specific range of block numbers?

Reply:
That would be nice but I don't see any way to do it without some fairly invasive changes to the interface. Let me know if you see a way to do this neatly.

Reply:
I though of something like this, just to limit the scope of the check:
```go
	var emptyHash common.Hash
	if head.UncleHash != emptyHash  && len(body.UncleHashes) == 0 && head.Number.Cmp(big.NewInt(12345)) < 0 {
		return nil, errors.New("server returned empty uncle list but block header indicates uncles")
	}
	```

Reply:
Ok, now I get it.
> The first blocks of the celo chain lacked the uncles field

It sounded like it was only for the first couple of blocks, not until gingerbread.